capella2polarion/elements/capella_object_renderer.py

```diff
@@ -234,23 +237,40 @@ def render_jinja_template(
         env = self._get_jinja_env(str(template_folder))
         template = env.get_template(str(template_path))
         rendered_jinja = template.render(
-            object=capella_element,
+            object=converter_data.capella_element,
             model=self.model,
-            work_item=work_item,
+            work_item=converter_data.work_item,
+            config=converter_data.type_config,
             **(render_params or {}),
         )
-        return self.sanitize_text(capella_element, rendered_jinja, errors)
+        return self.sanitize_text(
+            converter_data.capella_element,
+            rendered_jinja,
+            converter_data.errors,
+        )
 
     def setup_env(self, env: jinja2.Environment) -> None:
         """Add the link rendering filter."""
         env.filters["make_href"] = self.__make_href_filter
+        env.filters["render_diagram_with_params"] = (
+            self.__render_diagram_with_params
+        )
         env.globals["insert_diagram"] = self.__insert_diagram
```

Comment:
Since it works conceptually very similarly to `insert_diagram`, IMO it would make sense to also pass `render_diagram_with_params` as a callable global, not as a filter.